src/www/ui/api/index.php

```diff
@@ -273,12 +273,12 @@ function (\Slim\Routing\RouteCollectorProxy $app) {
   function (\Slim\Routing\RouteCollectorProxy $app) {
     $app->get('', GroupController::class . ':getGroups');
     $app->post('', GroupController::class . ':createGroup');
-    $app->post('/{id:\\d+}/user/{userId:\\d+}', GroupController::class . ':addMember');
-    $app->delete('/{id:\\d+}', GroupController::class . ':deleteGroup');
-    $app->delete('/{id:\\d+}/user/{userId:\\d+}', GroupController::class . ':deleteGroupMember');
+    $app->post('/{pathParam:[\\w\\- \\.\\d]+}/user/{userPathParam:[\\w\\- \\.\\d]+}', GroupController::class . ':addMember');
+    $app->delete('/{pathParam:[\\w\\- \\.\\d]+}', GroupController::class . ':deleteGroup');
+    $app->delete('/{pathParam:[\\w\\- \\.\\d]+}/user/{userPathParam:[\\w\\- \\.\\d]+}', GroupController::class . ':deleteGroupMember');
     $app->get('/deletable', GroupController::class . ':getDeletableGroups');
-    $app->get('/{id:\\d+}/members', GroupController::class . ':getGroupMembers');
-    $app->put('/{id:\\d+}/user/{userId:\\d+}', GroupController::class . ':changeUserPermission');
+    $app->get('/{pathParam:[\\w\\- \\.\\d]+}/members', GroupController::class . ':getGroupMembers');
+    $app->put('/{pathParam:[\\w\\- \\.\\d]+}/user/{userPathParam:[\\w\\- \\.\\d]+}', GroupController::class . ':changeUserPermission');
```

Comment:
I would rather not allow whitespace in the group/user name. Please remove the whitespace from the regex.

Reply:
Done